packages/substrate-client/src/archive/public-types.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  StorageItemInput,
+  StorageItemResponse,
+  StorageResult,
+} from "@/chainhead"
+import type { AbortablePromiseFn } from "../common-types"
+
+export type ArchiveStorageItemInput = StorageItemInput & {
+  paginationStartKey?: string
+}
+
+export interface Archive {
+  body: AbortablePromiseFn<[hash: string], Array<string>>
+  call: AbortablePromiseFn<
+    [hash: string, fnName: string, callParameters: string],
+    string
+  >
+  header: (hash: string) => Promise<string>
```

Comment:
Shouldn't this also be AbortablePromiseFn?

Reply:
I was on the fence, but yeah, let's do it.